sqldelight-compiler/src/main/kotlin/app/cash/sqldelight/core/compiler/model/BindableQuery.kt

```diff
@@ -75,14 +76,16 @@ abstract class BindableQuery(
    */
   val arguments: List<Argument> by lazy {
     if (statement is SqlInsertStmt && statement.acceptsTableInterface()) {
+      val offset = statement.insertStmtValues?.childOfType(SqlTypes.BIND_EXPR)!!.range.start
       return@lazy statement.columns.mapIndexed { index, column ->
         Argument(
-          index + 1,
-          column.type().let {
+          index = index + 1,
+          type = column.type().let {
             it.copy(
               name = "${allocateName(statement.tableName)}.${it.name}",
             )
           },
+          positions = mutableListOf(offset + 3 * index + 1),
```

Comment:
same here: why 3?

Reply:
This is due to the same logic. The single `?` in Insert statements replaced by the compiler is replaced by 3 characters per questionmark (excluding the first). I could add a comment to it explaining the calculation. Suggestions:

```
          // for every argument after the first there are 3 characters inserted. e.g.:
          // (Assuming a table with 2 columns)
          // `INSERT ... VALUES ?` is replaced by
          // `INSERT ... VALUES (?, ?)
          // the first argument starts at the offset of the original questionmark + 1, every following ? is further 
          // offset by 3 characters: the last questionmark, one comma and a space. Therefor the actual offset for each 
          // argument is calculated by <offset of the replaced questionmark> + 1 + 3 * <number of previous arguments>
          positions = mutableListOf(offset + 3 * index + 1),
```

Reply:
Yes, please add both comments